bundlegen/core/bundle_processor.py

```diff
@@ -536,29 +537,175 @@ def _process_logging(self):
         self.oci_config['rdkPlugins']['logging'] = logging_plugin
         return
 
+    def _remove_from_rootfs(self, rootfs_filepath):
+        """Remove from rootfs. If it is a link, also remove linked file.
+        """
+        link = os.path.realpath(rootfs_filepath)
+        os.remove(rootfs_filepath)
+        if (link != rootfs_filepath):
+            os.remove(link)
+
+    def _take_host_lib(self, srclib, dstlib, api_info):
+        """ The lib version from the host was choosen. Log it, create mount bind
+            and remove from image rootfs if present there.
+            Also for any sublibs if present.
+        """
+        logger.debug(f"HOST version choosen: {srclib}")
+        self.handled_libs.add(dstlib)
+        rootfs_filepath = os.path.join(self.rootfs_path, dstlib.lstrip('/'))
+        self._add_bind_mount(srclib, dstlib)
+        if os.path.exists(rootfs_filepath):
+            logger.debug(f"Removing from rootfs: {dstlib}")
+            self._remove_from_rootfs(rootfs_filepath)
+        if api_info and api_info.get('sublibs'):
+            for sublib in api_info['sublibs']:
+                logger.debug(f"HOST version choosen: {sublib}")
+                self.handled_libs.add(sublib)
+                self._add_bind_mount(sublib, sublib)
+                sublib_rootfs_filepath = os.path.join(self.rootfs_path, sublib.lstrip('/'))
+                if os.path.exists(sublib_rootfs_filepath):
+                    logger.debug(f"Removing from rootfs: {sublib}")
+                    self._remove_from_rootfs(sublib_rootfs_filepath)
+        if api_info and api_info.get('deps'):
+            for neededlib in api_info['deps']:
+                self._mount_or_use_rootfs(neededlib, neededlib)
+
+    def _determine_sublibs(self):
+        """
+        Determine sublibs inside libs info.
+        Specific processing for libc and its sublibs like libresolv.
+        First it determines the actual libc lib by finding the one
+        with the most GLIBC* versions inside its "Version definition section".
+        It also looks for other libs that only contain GLIBC* entries on their
+        "Version definition section". These are then assumed to be "sublibraries"
+        of libc.so
+        """
+        if not self.platform_cfg.get('libs'):
+            return
+
+        libs = self.platform_cfg['libs']
+        max = 0
+        libc = None
+        sublibs = []
+        for lib in libs:
+            m1 = re.match(r"^/lib/lib\S+\.so\.\d+$", lib['name'])
```

Comment:
This is only for find libc and on my platforms it seems always located under /lib. But perhaps on other it might be under /usr/lib. I will add that. 

Regarding the unversioned libs: yes that is intentional. Mainly because both /lib/libc-2.24.so and /lib/libc.so.6 are inside the _libs.json file and for finding libc I want to find  /lib/libc.so.6 and not /lib/libc-2.24.so...